lib/cosmos/streams/tcpip_socket_stream.rb

```diff
@@ -48,9 +50,9 @@ def read
       # No read mutex is needed because there is only one stream procesor
       # reading
       begin
-        data = @read_socket.recv_nonblock(65535)
-        @raw_logger_pair.read_logger.write(data) if @raw_logger_pair
-      rescue Errno::EAGAIN, Errno::EWOULDBLOCK
+        @read_socket.read_nonblock(65535, @data)
```

Comment:
I still don't know about this change.   It could break lots of use cases.    For example:
array = []
10.times do
  array << stream.read
end

Array ends up with the same mutated object 10 times.   I think a new string is really what is expected.